### PR TITLE
test: add integration test for v1

### PR DESCRIPTION
**Pull request by athakor**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-game-servers/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by athakor** [comment]
@chingor13 integration test failed due to  `Game Services API` is not `enable`. Please enable it to run the tests successfully.

File: pom.xml [hunk 1]
```diff
@@ -63,6 +63,7 @@
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     <github.global.server>github</github.global.server>
     <site.installationModule>google-cloud-game-servers-v1beta-parent</site.installationModule>
+    <google.core.version>1.92.5</google.core.version>
```
Comment (stephaniewang526) [inline pom.xml:66]: I don't think we need to manage the version here as we are using the [google-cloud-shared-dependencies-bom](https://github.com/googleapis/java-game-servers/blob/35e8ffcae005549887f55d771196939508e82ab6/pom.xml#L99) which provides a version for `core`

File: pom.xml [hunk 2]
```diff
@@ -107,6 +108,12 @@
         <version>4.13</version>
         <scope>test</scope>
       </dependency>
+      <dependency>
+        <groupId>com.google.cloud</groupId>
+        <artifactId>google-cloud-core</artifactId>
+        <version>${google.core.version}</version>
```
Comment (stephaniewang526) [inline pom.xml:114]: as noted above

**Comment by stephaniewang526** [comment]
Please whitelist the project indicated in [the error](https://source.cloud.google.com/results/invocations/622d7242-04c8-461c-8f49-ad4a44778cb2/targets/github%2Fjava-game-servers%2Fgoogle-cloud-game-servers%2Ftarget%2Ffailsafe-reports%2Fintegration%2Fcom.google.cloud.gaming.v1.it.ITSystemTest/tests) for IT to run.

**Comment by chingor13** [comment]
I enabled the API and the test now passes

File: pom.xml (hunk 1 above)
Comment (athakor) [reply pom.xml]: done

File: pom.xml (hunk 2 above)
Comment (athakor) [reply pom.xml]: done

**Comment by athakor** [comment]
@stephaniewang526 PTAL

**Pull request merged**